features/src/autogluon/features/generators/datetime.py

```diff
@@ -1,7 +1,10 @@
 import logging
-
+import os
+import holidays
```

Comment:
lazily import holidays within `_generate_features_datetime` and only when `self.holiday_encoding=True`, this way the user won't crash if they don't have holidays installed and they aren't using the holiday's functionality.